### PR TITLE
Clean csv

### PR DESCRIPTION
**Pull request by gnott**
A variant of CSV data observed recently where the data was spread across multiple lines with indents for each line in the XML-like output, shown here in the new ``datasets.csv`` test fixture. 

The concept for the fix was to clean the CSV data first, write more standard CSV data files, which can then be parsed by csvreader. To be most effective, It should have no side effect when it reads and writes normal CSV data.

The main feature here is the ``clean_csv()`` function and the test fixtures to test it. It is the result of a couple revisions and although it looks a little basic the tests are passing. Other changes are some tweaks and refactors.

The CSV data is back to normal, so it is not a rush to deploy this fix now. If we merge it in we should hopefully be protected if the multi-line data appears again.

### CODE REVIEW

File: parseCSVFiles.py [hunk 1]
```diff
@@ -59,6 +60,8 @@
 
 OVERFLOW_XLS_FILES = settings.OVERFLOW_XLS_FILES
 
+CSV_CLEAN_FILE_LIST = []
```
Comment (giorgiosironi) [inline parseCSVFiles.py]: this is global state, is there a way to make it smaller in scope, for example as a field of an object?

File: tests/test_data/clean_csv/datasets.csv [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        LTLTauthors_text_listGTGTCembrowski M, Spruston N,LTLT/authors_text_listGTGT
+        LTLTidGTGThttp://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67403LTLT/idGTGT
+        LTLTlicense_infoGTGTGSE67403LTLT/license_infoGTGT
+        LTLTtitleGTGTSpatial gene expression gradients underlie prominent heterogeneity of CA1 pyramidal neuronsLTLT/titleGTGT
+        LTLTyearGTGT2016LTLT/yearGTGT
+      LTLT/datasetGTGT
+    LTLTdatasets_indGTGT1LTLT/datasets_indGTGT
+  LTLT/prev_published_datasetsGTGT
+LTLT/xmlGTGT
+"
```
Comment (giorgiosironi) [inline tests/test_data/clean_csv/datasets.csv]: the set of fixtures is extensive :+1:

File: parseCSVFiles.py [hunk 3]
```diff
@@ -91,12 +94,52 @@ def get_xls_path(path_type):
...
     return path
 
+def clean_csv(path):
+    "fix CSV file oddities making it difficult to parse"
+    clean_csv_data = ''
+    new_path = os.path.join(settings.TMP_DIR, path.split(os.sep)[-1])
+    # Return the cleaned file name if it is already cleaned
+    if path in CSV_CLEAN_FILE_LIST:
+        return new_path
+    with open(path, 'rb') as open_read_file:
```
[the remark this answers is not in the document]
Comment (gnott) [reply parseCSVFiles.py]: I agree, once I started to write tests, I didn't expect to have to load a file to test the meat of the function. I will separate the file read/write from the cleaning logic.

[the remark this answers is not in the document]
Comment (gnott) [reply parseCSVFiles.py]: The one function is split into four now, with some test cases for each of them. Not to say it makes it easier to understand but they are hopefully easier to test and diagnose if something goes wrong.

File: parseCSVFiles.py (hunk 1 above)
Comment (gnott) [reply parseCSVFiles.py]: The globals are not ideal. What I'm hoping is this particular library will not be around much longer, and may these can be ignored for now? Improvements can be made in the newer https://github.com/elifesciences/ejp-csv-parser/blob/develop/ejpcsvparser/csv_data.py  module of the other parsing library, where there are still some globals but some improvement in how the settings are specified, with work still to be done.

**Pull request merged**